lib/internal/worker/everysync/index.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const {
+  AtomicsNotify,
+  AtomicsStore,
+  AtomicsWait,
+  AtomicsWaitAsync,
+  Int32Array,
+  ObjectKeys,
+} = primordials;
+
+const {
+  codes: {
+    ERR_WORKER_MESSAGING_TIMEOUT,
+  },
+} = require('internal/errors');
+
+const { read, write } = require('internal/worker/everysync/objects');
+const {
+  OFFSET,
+  TO_MAIN,
+  TO_WORKER,
+} = require('internal/worker/everysync/indexes');
+
+/**
+ * Creates a synchronous API facade from a shared memory buffer.
+ * This function is meant to be used in the main thread to communicate with
+ * a worker thread that has called `wire()` on the same shared memory.
+ * @param {SharedArrayBuffer} data - The shared memory buffer for communication
+ * @param {object} [opts={}] - Options object
+ * @param {number} [opts.timeout=1000] - Timeout in milliseconds for synchronous operations
+ * @returns {object} - An object with methods that match the ones exposed by the worker
+ */
+function makeSync(data, opts = {}) {
+  const timeout = opts.timeout || 1000;
+  const metaView = new Int32Array(data);
+
+  const res = AtomicsWait(metaView, TO_WORKER, 0, timeout);
+  AtomicsStore(metaView, TO_WORKER, 0);
+
+  if (res === 'ok') {
+    const obj = read(data, OFFSET);
+
+    const api = {};
+    for (const key of obj) {
```

Comment:
Just curious, why this pattern and not a `Proxy`? I know `Proxy` objects have some perf downsides but this would seem a tailor made use case for it.

Reply:
Indeed. I think I would actually significantly simplify this.